src/services/identity/IIdentity.ts

```diff
@@ -1,42 +1,126 @@
 import type {
-  CredentialPayload,
-  DIDDocument,
-  DIDResolutionResult,
-  IIdentifier,
-  IVerifyResult,
-  ManagedKeyInfo,
-  TAgent,
-  VerifiableCredential,
-  VerifiablePresentation,
+	CredentialPayload, DIDDocument, DIDResolutionResult, IIdentifier, IVerifyResult,
+	ManagedKeyInfo, TAgent, VerifiableCredential, VerifiablePresentation,
 } from '@veramo/core'
 import type { AbstractPrivateKeyStore } from '@veramo/key-manager'
 import type { ResourcePayload } from '@cheqd/did-provider-cheqd'
-import type { BulkRevocationResult, BulkSuspensionResult, BulkUnsuspensionResult, CreateStatusList2021Result, RevocationResult, StatusCheckResult, SuspensionResult, UnsuspensionResult } from '@cheqd/did-provider-cheqd/build/types/agent/ICheqd'
-import type { BroadCastStatusListOptions, CheckStatusListOptions, CreateStatusListOptions, CredentialRequest, StatusOptions, UpdateStatusListOptions, VeramoAgent, VerificationOptions } from '../../types/types'
+import type {
+	BulkRevocationResult, BulkSuspensionResult, BulkUnsuspensionResult, CreateStatusList2021Result,
+	RevocationResult, StatusCheckResult, SuspensionResult, UnsuspensionResult,
+} from '@cheqd/did-provider-cheqd/build/types/agent/ICheqd'
+import type {
+	BroadCastStatusListOptions, CheckStatusListOptions, CreateStatusListOptions, CredentialRequest,
+	StatusOptions, UpdateStatusListOptions, VeramoAgent, VerificationOptions,
+} from '../../types/types'
 
 export interface IIdentity {
-  agent?: TAgent<any>
```

Comment:
Why did you remove `agent` and `privateStore` from interface?

Reply:
If I move it `IIdentity` interface that I need to redefine it `AbstractIdentity` class.